app/controllers/manageiq/api/base_controller/blueprints.rb

```diff
@@ -42,17 +42,33 @@ def update_bundle(blueprint, bundle)
         def deserialize_bundle(bundle)
           options = {}
           if bundle["service_catalog"]
-            options[:service_catalog] = ServiceTemplateCatalog.find(parse_id(bundle["service_catalog"], :service_catalogs))
+            options["service_catalog"] = service_catalog_option(bundle)
           end
           if bundle["service_dialog"]
-            options[:service_dialog] = Dialog.find(parse_id(bundle["service_dialog"], :service_dialogs))
+            options[:service_dialog] = service_dialog_option(bundle)
           end
           options[:service_templates] = bundle.fetch("service_templates", []).collect do |st|
             ServiceTemplate.find(parse_id(st, :service_templates))
           end
           options[:entry_points] = bundle["automate_entrypoints"] if bundle["automate_entrypoints"]
           options
         end
+
+        def service_catalog_option(bundle)
+          if bundle["service_catalog"].blank?
```

Comment:
I wonder if this would read better with the happy path first? i.e.

```ruby
if bundle["service_catalog"]
  ServiceTemplateCatalog.find(etc)
else
  nil
end
```

Since `nil` is the implied `else` in an `if` statement, it could be further condensed:

```ruby
if bundle["service_catalog"]
  ServiceTemplateCatalog.find(etc)
end
```

WDYT?